### ISSUE TITLE
Timeouts after update from 2.4.7 to 2.4.8 on Debian 11

### ISSUE THREAD

**Issue by takenek** [haproxy/haproxy#1448]
### Detailed Description of the Problem

After update haproxy from 2.4.7 to 2.4.8 on Debian 11 clients report a timeouts.
In logs i saw strange increase of connection on front-end:
2.4.7:
Nov  8 22:48:00 HP1 haproxy[643136]: 192.168.1.10:55054 [08/Nov/2021:22:47:59.939] XML XML/XML4 0/0/123 46811 -- 356/2/1/0/0 0/0

2.4.8:
Nov  8 22:50:00 HP1 haproxy[645691]: 192.168.1.10:37550 [08/Nov/2021:22:49:59.993] XML XML/XML4 0/0/60 33369 -- 344/78/2/0/0 0/0

Like you see there is 78 connection to frontend and all the time is verry high on 2.4.8: 344/78/2/0/0 0/0 was on 2.4.7: 356/2/1/0/0 0/0

In haproxy stats no errors down nodes or anything everything green.

### Expected Behavior

No timeouts

### Steps to Reproduce the Behavior

1. Just update from 2.4.7 to 2.4.8 via https://haproxy.debian.net/

### What is your configuration?

```haproxy
global
        log /dev/log    local0
        log /dev/log    local1 notice
        chroot /var/lib/haproxy
        stats socket /run/haproxy/admin.sock mode 660 level admin expose-fd listeners
        stats timeout 30s
        user haproxy
        group haproxy
        daemon

        # Default SSL material locations
        ca-base /etc/ssl/certs
        crt-base /etc/ssl/private

        # See: https://ssl-config.mozilla.org/#server=haproxy&server-version=2.0.3&config=intermediate
        ssl-default-bind-ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384
        ssl-default-bind-ciphersuites TLS_AES_128_GCM_SHA256:TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256
        ssl-default-bind-options ssl-min-ver TLSv1.2 no-tls-tickets

defaults
        log     global
        mode    http
        option  tcplog
        option  dontlognull
        timeout connect 5000
        timeout client  50000
        timeout server  50000
        errorfile 400 /etc/haproxy/errors/400.http
        errorfile 403 /etc/haproxy/errors/403.http
        errorfile 408 /etc/haproxy/errors/408.http
        errorfile 500 /etc/haproxy/errors/500.http
        errorfile 502 /etc/haproxy/errors/502.http
        errorfile 503 /etc/haproxy/errors/503.http
        errorfile 504 /etc/haproxy/errors/504.http

# XML
        listen  XML
        bind    192.168.1.1:80
        mode    http
        option  httpchk /xml.html
        option  persist
        option  forwardfor
        balance roundrobin
        cookie  XML insert
        maxconn 2000
        fullconn 2000
        server  XML1 10.10.10.2:80 cookie XML1 check inter 4000 fall 3 maxconn 500
        server  XML2 10.10.10.3:80 cookie XML2 check inter 4000 fall 3 maxconn 500
        server  XML3 10.10.10.4:80 cookie XML3 check inter 4000 fall 3 maxconn 500
        server  XML4 10.10.10.5:80 cookie XML4 check inter 4000 fall 3 maxconn 500
        timeout server 20000
        timeout client 150000
        timeout connect 40000

# STATS
        listen stats
        bind 192.168.1.2:65080
        mode http
        stats enable
        stats uri /
        maxconn 400
        fullconn 400
        stats auth login:password
        stats refresh 10s
```

### Output of `haproxy -vv`

```plain
HAProxy version 2.4.8-1~bpo11+1 2021/11/04 - https://haproxy.org/
Status: long-term supported branch - will stop receiving fixes around Q2 2026.
Known bugs: http://www.haproxy.org/bugs/bugs-2.4.8.html
Running on: Linux 5.10.0-9-amd64 #1 SMP Debian 5.10.70-1 (2021-09-30) x86_64
Build options :
  TARGET  = linux-glibc
  CPU     = generic
  CC      = cc
  CFLAGS  = -O2 -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -Wall -Wextra -Wdeclaration-after-statement -fwrapv -Wno-address-of-packed-member -Wno-unused-label -Wno-sign-compare -Wno-unused-parameter -Wno-clobbered -Wno-missing-field-initializers -Wno-cast-function-type -Wtype-limits -Wshift-negative-value -Wshift-overflow=2 -Wduplicated-cond -Wnull-dereference
  OPTIONS = USE_PCRE2=1 USE_PCRE2_JIT=1 USE_OPENSSL=1 USE_LUA=1 USE_SLZ=1 USE_SYSTEMD=1 USE_PROMEX=1
  DEBUG   =

Feature list : +EPOLL -KQUEUE +NETFILTER -PCRE -PCRE_JIT +PCRE2 +PCRE2_JIT +POLL -PRIVATE_CACHE +THREAD -PTHREAD_PSHARED +BACKTRACE -STATIC_PCRE -STATIC_PCRE2 +TPROXY +LINUX_TPROXY +LINUX_SPLICE +LIBCRYPT +CRYPT_H +GETADDRINFO +OPENSSL +LUA +FUTEX +ACCEPT4 -CLOSEFROM -ZLIB +SLZ +CPU_AFFINITY +TFO +NS +DL +RT -DEVICEATLAS -51DEGREES -WURFL +SYSTEMD -OBSOLETE_LINKER +PRCTL -PROCCTL +THREAD_DUMP -EVPORTS -OT -QUIC +PROMEX -MEMORY_PROFILING

Default settings :
  bufsize = 16384, maxrewrite = 1024, maxpollevents = 200

Built with multi-threading support (MAX_THREADS=64, default=4).
Built with OpenSSL version : OpenSSL 1.1.1k  25 Mar 2021
Running on OpenSSL version : OpenSSL 1.1.1k  25 Mar 2021
OpenSSL library supports TLS extensions : yes
OpenSSL library supports SNI : yes
OpenSSL library supports : TLSv1.0 TLSv1.1 TLSv1.2 TLSv1.3
Built with Lua version : Lua 5.3.3
Built with the Prometheus exporter as a service
Built with network namespace support.
Built with libslz for stateless compression.
Compression algorithms supported : identity("identity"), deflate("deflate"), raw-deflate("deflate"), gzip("gzip")
Built with transparent proxy support using: IP_TRANSPARENT IPV6_TRANSPARENT IP_FREEBIND
Built with PCRE2 version : 10.36 2020-12-04
PCRE2 library supports JIT : yes
Encrypted password support via crypt(3): yes
Built with gcc compiler version 10.2.1 20210110

Available polling systems :
      epoll : pref=300,  test result OK
       poll : pref=200,  test result OK
     select : pref=150,  test result OK
Total: 3 (3 usable), will use epoll.

Available multiplexer protocols :
(protocols marked as <default> cannot be specified using 'proto' keyword)
              h2 : mode=HTTP       side=FE|BE     mux=H2       flags=HTX|CLEAN_ABRT|HOL_RISK|NO_UPG
            fcgi : mode=HTTP       side=BE        mux=FCGI     flags=HTX|HOL_RISK|NO_UPG
       <default> : mode=HTTP       side=FE|BE     mux=H1       flags=HTX
              h1 : mode=HTTP       side=FE|BE     mux=H1       flags=HTX|NO_UPG
       <default> : mode=TCP        side=FE|BE     mux=PASS     flags=
            none : mode=TCP        side=FE|BE     mux=PASS     flags=NO_UPG

Available services : prometheus-exporter
Available filters :
        [SPOE] spoe
        [CACHE] cache
        [FCGI] fcgi-app
        [COMP] compression
        [TRACE] trace
```

**Comment by capflam** [comment]
Could you share the redacted output of the `show sess all` command please ? You can sent it me privately if you prefer. It could be a bit huge, but only sessions with a high 'age' time are relevant.

**Comment by takenek** [comment]
I forgot give haproxy -vv from 2.4.7 ( this one working without any issue )

HAProxy version 2.4.7-1~bpo11+1 2021/10/07 - https://haproxy.org/
Status: long-term supported branch - will stop receiving fixes around Q2 2026.
Known bugs: http://www.haproxy.org/bugs/bugs-2.4.7.html
Running on: Linux 5.10.0-9-amd64 #1 SMP Debian 5.10.70-1 (2021-09-30) x86_64
Build options :
  TARGET  = linux-glibc
  CPU     = generic
  CC      = cc
  CFLAGS  = -O2 -g -O2 -ffile-prefix-map=/build/haproxy-2.4.7=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -Wall -Wextra -Wdeclaration-after-statement -fwrapv -Wno-address-of-packed-member -Wno-unused-label -Wno-sign-compare -Wno-unused-parameter -Wno-clobbered -Wno-missing-field-initializers -Wno-cast-function-type -Wtype-limits -Wshift-negative-value -Wshift-overflow=2 -Wduplicated-cond -Wnull-dereference
  OPTIONS = USE_PCRE2=1 USE_PCRE2_JIT=1 USE_OPENSSL=1 USE_LUA=1 USE_SLZ=1 USE_SYSTEMD=1 USE_PROMEX=1
  DEBUG   =

Feature list : +EPOLL -KQUEUE +NETFILTER -PCRE -PCRE_JIT +PCRE2 +PCRE2_JIT +POLL -PRIVATE_CACHE +THREAD -PTHREAD_PSHARED +BACKTRACE -STATIC_PCRE -STATIC_PCRE2 +TPROXY +LINUX_TPROXY +LINUX_SPLICE +LIBCRYPT +CRYPT_H +GETADDRINFO +OPENSSL +LUA +FUTEX +ACCEPT4 -CLOSEFROM -ZLIB +SLZ +CPU_AFFINITY +TFO +NS +DL +RT -DEVICEATLAS -51DEGREES -WURFL +SYSTEMD -OBSOLETE_LINKER +PRCTL -PROCCTL +THREAD_DUMP -EVPORTS -OT -QUIC +PROMEX -MEMORY_PROFILING

Default settings :
  bufsize = 16384, maxrewrite = 1024, maxpollevents = 200

Built with multi-threading support (MAX_THREADS=64, default=4).
Built with OpenSSL version : OpenSSL 1.1.1k  25 Mar 2021
Running on OpenSSL version : OpenSSL 1.1.1k  25 Mar 2021
OpenSSL library supports TLS extensions : yes
OpenSSL library supports SNI : yes
OpenSSL library supports : TLSv1.0 TLSv1.1 TLSv1.2 TLSv1.3
Built with Lua version : Lua 5.3.3
Built with the Prometheus exporter as a service
Built with network namespace support.
Built with libslz for stateless compression.
Compression algorithms supported : identity("identity"), deflate("deflate"), raw-deflate("deflate"), gzip("gzip")
Built with transparent proxy support using: IP_TRANSPARENT IPV6_TRANSPARENT IP_FREEBIND
Built with PCRE2 version : 10.36 2020-12-04
PCRE2 library supports JIT : yes
Encrypted password support via crypt(3): yes
Built with gcc compiler version 10.2.1 20210110

Available polling systems :
      epoll : pref=300,  test result OK
       poll : pref=200,  test result OK
     select : pref=150,  test result OK
Total: 3 (3 usable), will use epoll.

Available multiplexer protocols :
(protocols marked as <default> cannot be specified using 'proto' keyword)
              h2 : mode=HTTP       side=FE|BE     mux=H2       flags=HTX|CLEAN_ABRT|HOL_RISK|NO_UPG
            fcgi : mode=HTTP       side=BE        mux=FCGI     flags=HTX|HOL_RISK|NO_UPG
       <default> : mode=HTTP       side=FE|BE     mux=H1       flags=HTX
              h1 : mode=HTTP       side=FE|BE     mux=H1       flags=HTX|NO_UPG
       <default> : mode=TCP        side=FE|BE     mux=PASS     flags=
            none : mode=TCP        side=FE|BE     mux=PASS     flags=NO_UPG

Available services : prometheus-exporter
Available filters :
        [SPOE] spoe
        [CACHE] cache
        [FCGI] fcgi-app
        [COMP] compression
        [TRACE] trace

**Comment by takenek** [comment]
2.4.7 working good:

```
0x7f633402aaf0: [09/Nov/2021:11:10:44.986911] id=624297 proto=unix_stream source=unix:1
  flags=0x8, conn_retries=0, srv_conn=(nil), pend_pos=(nil) waiting=0 epoch=0x1
  frontend=GLOBAL (id=0 mode=tcp), listener=? (id=1) addr=unix:1
  backend=<NONE> (id=-1 mode=-)
  server=<NONE> (id=-1)
  task=0x7f6334162990 (state=0x10000 nice=-64 calls=1 rate=0 exp=21s tmask=0x2 age=9s)
  si[0]=0x7f633402ada0 (state=EST flags=0x200008 endp0=CS:0x7f6324121bf0 exp=<NEVER> et=0x000 sub=0)
  si[1]=0x7f633402adf8 (state=EST flags=0x204018 endp1=APPCTX:0x7f63340fb430 exp=<NEVER> et=0x000 sub=0)
  co0=0x555ad818fde0 ctrl=unix_stream xprt=RAW mux=PASS data=STRM target=LISTENER:0x555ad7eace60
      flags=0x00003300 fd=318 fd.state=120 updt=0 fd.tmask=0x2
      cs=0x7f6324121bf0 csf=0x00008200 ctx=(nil)
  app1=0x7f63340fb430 st0=9 st1=1 st2=3 applet=<CLI> tmask=0x2 nice=-64 calls=3 rate=0 cpu=0 lat=0
  req=0x7f633402ab00 (f=0xc48202 an=0x0 pipe=0 tofwd=-1 total=21)
      an_exp=<NEVER> rex=30s wex=<NEVER>
      buf=0x7f633402ab08 data=0x7f633404c870 o=0 p=0 i=0 size=16384
  res=0x7f633402ab60 (f=0x80048202 an=0x0 pipe=0 tofwd=-1 total=3)
      an_exp=<NEVER> rex=<NEVER> wex=<NEVER>
      buf=0x7f633402ab68 data=0x7f633403d3b0 o=0 p=0 i=0 size=16384

0x7f632c035ca0: [09/Nov/2021:11:10:52.978368] id=624702 proto=tcpv4 source=aaa.aaa.aaa.aaa:54486
  flags=0x84e, conn_retries=3, srv_conn=0x555ad7f12160, pend_pos=(nil) waiting=0 epoch=0x1
  frontend=XML (id=9 mode=http), listener=? (id=1) addr=bbb.bbb.bbb.bbb:80
  backend=XML (id=9 mode=http) addr=ccc.ccc.ccc.ccc:34602
  server=XML1 (id=3) addr=ddd.ddd.ddd.ddd:80
  task=0x7f632c0a8ca0 (state=0x10000 nice=0 calls=2 rate=2 exp=19s tmask=0x4 age=1s)
  txn=0x555ad8382080 flags=0x0 meth=3 status=-1 req.st=MSG_DONE rsp.st=MSG_RPBEFORE req.f=0x05 rsp.f=0x00
  si[0]=0x7f632c035f50 (state=EST flags=0x200008 endp0=CS:0x7f632407c970 exp=<NEVER> et=0x000 sub=1)
  si[1]=0x7f632c035fa8 (state=EST flags=0x200118 endp1=CS:0x7f632413eb30 exp=<NEVER> et=0x000 sub=1)
  co0=0x555ad8262450 ctrl=tcpv4 xprt=RAW mux=H1 data=STRM target=LISTENER:0x555ad7f0ff00
      flags=0x80003300 fd=390 fd.state=20 updt=0 fd.tmask=0x4
      cs=0x7f632407c970 csf=0x00004000 ctx=0x555ad827d200
  co1=0x555ad8130bf0 ctrl=tcpv4 xprt=RAW mux=H1 data=STRM target=SERVER:0x555ad7f12160
      flags=0x00003300 fd=500 fd.state=10421 updt=0 fd.tmask=0x4
      cs=0x7f632413eb30 csf=0x00000000 ctx=0x555ad83aa510
  req=0x7f632c035cb0 (f=0x48c40080 an=0x8000 pipe=0 tofwd=0 total=370)
      an_exp=<NEVER> rex=<NEVER> wex=<NEVER>
      buf=0x7f632c035cb8 data=(nil) o=0 p=0 i=0 size=0
      htx=0x555ad7a16080 flags=0x0 size=0 data=0 used=0 wrap=NO extra=0
  res=0x7f632c035d10 (f=0x80400000 an=0x1400000 pipe=0 tofwd=0 total=0)
      an_exp=<NEVER> rex=19s wex=<NEVER>
      buf=0x7f632c035d18 data=(nil) o=0 p=0 i=0 size=0
      htx=0x555ad7a16080 flags=0x0 size=0 data=0 used=0 wrap=NO extra=0
```
      
2.4.8 with issue:
```
0x562f6de12220: [09/Nov/2021:11:12:01.442642] id=1023 proto=unix_stream source=unix:1
  flags=0x8, conn_retries=0, srv_conn=(nil), pend_pos=(nil) waiting=0 epoch=0
  frontend=GLOBAL (id=0 mode=tcp), listener=? (id=1) addr=unix:1
  backend=<NONE> (id=-1 mode=-)
  server=<NONE> (id=-1)
  task=0x562f6de16860 (state=0x00 nice=-64 calls=1 rate=0 exp=23s tmask=0x1 age=6s)
  si[0]=0x562f6de124d0 (state=EST flags=0x200008 endp0=CS:0x562f6de4a600 exp=<NEVER> et=0x000 sub=0)
  si[1]=0x562f6de12528 (state=EST flags=0x204018 endp1=APPCTX:0x562f6de04e70 exp=<NEVER> et=0x000 sub=0)
  co0=0x7f1270063eb0 ctrl=unix_stream xprt=RAW mux=PASS data=STRM target=LISTENER:0x562f6db96e60
      flags=0x00003300 fd=344 fd.state=120 updt=0 fd.tmask=0x1
      cs=0x562f6de4a600 csf=0x00008200 ctx=0x32373030322e4131
  app1=0x562f6de04e70 st0=9 st1=1 st2=3 applet=<CLI> tmask=0x1 nice=-64 calls=3 rate=0 cpu=0 lat=0
  req=0x562f6de12230 (f=0xc48202 an=0x0 pipe=0 tofwd=-1 total=21)
      an_exp=<NEVER> rex=30s wex=<NEVER>
      buf=0x562f6de12238 data=0x562f6de0b190 o=0 p=0 i=0 size=16384
  res=0x562f6de12290 (f=0x80048202 an=0x0 pipe=0 tofwd=-1 total=3)
      an_exp=<NEVER> rex=<NEVER> wex=<NEVER>
      buf=0x562f6de12298 data=0x562f6dcd1a60 o=0 p=0 i=0 size=16384
	  
0x7f128002f420: [09/Nov/2021:11:12:07.525974] id=1341 proto=tcpv4 source=aaa.aaa.aaa.aaa:34364
  flags=0x84e, conn_retries=3, srv_conn=0x562f6dbfa320, pend_pos=(nil) waiting=0 epoch=0
  frontend=XML (id=9 mode=http), listener=? (id=1) addr=bbb.bbb.bbb.bbb:80
  backend=XML (id=9 mode=http) addr=ccc.ccc.ccc.ccc:40624
  server=XML1 (id=1) addr=ddd.ddd.ddd.ddd:80
  task=0x7f1280077f90 (state=0x00 nice=0 calls=2 rate=2 exp=19s tmask=0x2 age=0s)
  txn=0x7f128006d050 flags=0x0 meth=3 status=-1 req.st=MSG_DONE rsp.st=MSG_RPBEFORE req.f=0x05 rsp.f=0x00
  si[0]=0x7f128002f6d0 (state=EST flags=0x200008 endp0=CS:0x7f1280092940 exp=<NEVER> et=0x000 sub=1)
  si[1]=0x7f128002f728 (state=EST flags=0x200118 endp1=CS:0x7f12800854d0 exp=<NEVER> et=0x000 sub=1)
  co0=0x562f6de39030 ctrl=tcpv4 xprt=RAW mux=H1 data=STRM target=LISTENER:0x562f6dbf9f00
      flags=0x80003300 fd=335 fd.state=120 updt=0 fd.tmask=0x2
      cs=0x7f1280092940 csf=0x00004000 ctx=0x7f128006add0
  co1=0x7f12800921b0 ctrl=tcpv4 xprt=RAW mux=H1 data=STRM target=SERVER:0x562f6dbfa320
      flags=0x00003300 fd=350 fd.state=10421 updt=0 fd.tmask=0x2
      cs=0x7f12800854d0 csf=0x00000000 ctx=0x7f128008bf50
  req=0x7f128002f430 (f=0x48c40080 an=0x8000 pipe=0 tofwd=0 total=370)
      an_exp=<NEVER> rex=<NEVER> wex=<NEVER>
      buf=0x7f128002f438 data=(nil) o=0 p=0 i=0 size=0
      htx=0x562f6cc9a080 flags=0x0 size=0 data=0 used=0 wrap=NO extra=0
  res=0x7f128002f490 (f=0x80400000 an=0x1400000 pipe=0 tofwd=0 total=0)
      an_exp=<NEVER> rex=19s wex=<NEVER>
      buf=0x7f128002f498 data=(nil) o=0 p=0 i=0 size=0
      htx=0x562f6cc9a080 flags=0x0 size=0 data=0 used=0 wrap=NO extra=0
```

**Comment by takenek** [comment]
This is whole output of show sess all.

**Comment by capflam** [comment]
This means there are only 2 sessions on the 2.4.8 while 78 or so are reported in logs, right ?

**Comment by takenek** [comment]
No i just generate this command output.
Logs are from yesterday and show sess all i generate now.

**Comment by takenek** [comment]
This is log is almost from the same time like show sess all command output ( look at date/time ):

2.4.7 - working good:
Nov  9 11:10:45 HP haproxy[693455]: XXX.XXX.XXX.XXX:53474 [09/Nov/2021:11:10:44.957] XML XML/XML3 0/0/96 65154 -- 584/1/0/0/0 0/0

2.4.8 - with issue:
Nov  9 11:12:01 HP haproxy[728739]: XXX.XXX.XXX.XXX:33772 [09/Nov/2021:11:12:01.468] XML XML/XML3 0/0/62 41386 -- 273/45/4/0/0 0/0

**Comment by capflam** [comment]
So what is the current `feconn` counter value in your logs ? Because you have only one HTTP session (the other session is the cli one). If clients report timeout, it suggests some sessions are stalled for several seconds. It should be visible on the `show sess all` output. Or I misinterpreted the issue.

**Comment by capflam** [comment]
Ok thanks for the logs :)

**Comment by capflam** [comment]
Sorry to bother you. Could you share the `show fd` output now ? :)

**Comment by takenek** [comment]
@capflam Check Your mailbox ;-)

**Comment by takenek** [comment]
@capflam Check Your mailbox again :-)

**Comment by capflam** [comment]
Thanks @takenek. Well, after discussing about your issue with Willy, it appears that everything seems to work properly. In fact, in the 2.4.8, the H1 multiplexer was fixed to perform clean shutdown when client connections are closed. It was mandatory for the SSL (#1297). But at the end, this fixed a long time design issue about shutdowns.

So it is possible that this fix exhibits an hidden behavior on your platform. If you have slow clients or clients with a bad connectivity, this may explain what you observed, especially because you have "huge" client timeout. You may try to set `client-fin` timeout. For instance to 5s. It should reduce the number connections waiting for client shutdown. 

Of course, I may have wrong. There is a related issue (#1450). I proposed a fix for this issue, and it may also help you if I missed something.

It may  also be good to monitor TCP connections with netstat or ss.

**Comment by takenek** [comment]
I answer you on e-mail :-)

**Comment by wtarreau** [comment]
@takenek, please keep the discussion here. It's impossible for others to follow the exchanges and it seriously hurts the whole troubleshooting process to switch to private. It prevents others from sharing a "me too" and it prevents other developers from giving an opinion. Thanks.

**Comment by takenek** [comment]
@wtarreau we don't discussion on mail - i share there private files to diagnose problem.

**Comment by capflam** [comment]
Willy and I found several bugs that may explain the issue. Everything was pushed in the 2.4 repository. However, here is the 3 required patches for the 2.4 that should hopefully fix the issue:

* [0001-BUG-MEDIUM-conn-stream-Don-t-reset-CS-flags-on-close.patch.txt](https://github.com/haproxy/haproxy/files/7539262/0001-BUG-MEDIUM-conn-stream-Don-t-reset-CS-flags-on-close.patch.txt)
* [0002-BUG-MEDIUM-connection-make-cs_shutr-cs_shutw-cs_clos.patch.txt](https://github.com/haproxy/haproxy/files/7539263/0002-BUG-MEDIUM-connection-make-cs_shutr-cs_shutw-cs_clos.patch.txt)
* [0003-BUG-MEDIUM-mux-h1-Handle-delayed-silent-shut-in-h1_p.patch.txt](https://github.com/haproxy/haproxy/files/7539264/0003-BUG-MEDIUM-mux-h1-Handle-delayed-silent-shut-in-h1_p.patch.txt)

Could you try them please ?

**Comment by capflam** [comment]
@takenek, any news ? I'm sorry to bother you but we don't want to blindingly emit a new release. It is pretty hard to reproduce this kind of issues. So we need some help on this point to validate or invalidate fixes. Thanks.

**Comment by capflam** [comment]
FYI, the 2.4.9 was released.

**Comment by capflam** [comment]
I'm closing this issue because patches were backported everywhere. If the issue is not fixed with the 2.4.9, feel free to reopen it. Thanks !

**Comment by takenek** [comment]
Hey,
I can confirm version 2.4.9 working properly.

Best Regards
TaKeN
